sphinx/writers/text.py

```diff
@@ -29,6 +31,235 @@
+      this overwrite an existing cell if any.
+
+    - Either with relative positions: call the add_row() and
+      add_cell(Cell(...)) as needed.
+
+    Cell spanning on multiple rows or multiple columns (having a
+    colspan or rowspan greater than one) are automatically referenced
+    by all the table cells they covers. This is a usefull
+    representation as we can simply check `if self[x, y] is self[x,
+    y+1]` to recognize a rowspan.
+
+    Colwidth is not automatically computed, it has to be given, either
+    at construction time, either during the table construction.
+
+    Example usage:
+    table = Table([6, 6])
+    table.add_cell(Cell("foo"))
+    table.add_cell(Cell("bar"))
+    table.set_separator()
+    table.add_row()
+    table.add_cell(Cell("FOO"))
+    table.add_cell(Cell("BAR"))
+    print(str(table))
+    +--------+--------+
+    | foo    | bar    |
+    |========|========|
+    | FOO    | BAR    |
+    +--------+--------+
+
+    """
+    def __init__(self, colwidth=None):
+        self.lines = []
+        self.separator = 0
+        if colwidth is None:
+            self.colwidth = []
+        else:
+            self.colwidth = colwidth
+        self.current_line = 0
+        self.current_col = 0
+
+    def add_row(self):
+        """Add a row to the table, to use with add_cell().  It is not needed
+        to call add_row() before the first add_cell().
+        """
+        self.current_line += 1
+        self.current_col = 0
+
+    def set_separator(self):
+        """Sets the separator below the current line.
+        """
+        self.separator = len(self.lines)
+
+    def add_cell(self, cell):
+        """Add a cell to the current line, to use with add_row().  To add a
+        cell spanning on multiple lines or rows, simply set the
+        cell.colspan or cell.rowspan BEFORE inserting it to the table.
+        """
+        while self[self.current_line, self.current_col]:
+            self.current_col += 1
+        self[self.current_line, self.current_col] = cell
+        self.current_col += cell.colspan
+
+    def __getitem__(self, pos):
+        line, col = pos
+        self._ensure_has_line(line + 1)
+        self._ensure_has_column(col + 1)
+        return self.lines[line][col]
+
+    def __setitem__(self, pos, cell):
+        line, col = pos
+        self._ensure_has_line(line + cell.rowspan)
+        self._ensure_has_column(col + cell.colspan)
+        for dline in range(cell.rowspan):
+            for dcol in range(cell.colspan):
+                self.lines[line + dline][col + dcol] = cell
+                cell.row = line
+                cell.col = col
+
+    def _ensure_has_line(self, line):
+        while len(self.lines) < line:
+            self.lines.append([])
+
+    def _ensure_has_column(self, col):
+        for line in self.lines:
+            while len(line) < col:
+                line.append(None)
+
+    def __repr__(self):
+        out = []
```

Comment:
use a generator expression, it's more efficient. 
```
return '\n'.join(repr(line) for line in self.lines)
```

Reply:
👍